### ISSUE TITLE
Currently installed version detechtion incorrect for Android Studio Canary

### ISSUE THREAD

**Issue by louis993546** [microsoft/winget-cli#2803]
### Brief description of your issue

I have Android studio canary 2022.2.1.10 installed
But it seems like winget thought it is 2022.2.1.1

![image](https://user-images.githubusercontent.com/3873011/209593983-24daa278-730a-4603-9e96-0c5fba5c69a6.png)

Installed the local manifest of 2022.2.1.10 via
https://github.com/microsoft/winget-pkgs/pull/92304

### Steps to reproduce

1. Install Android Studio Canary 2022.2.1.10
2. run `winget upgrade`

### Expected behavior

It don't show 2022.2.1.10 as an outdated package

### Actual behavior

It mark there is a newer version (2022.2.1.9) available for upgrade

### Environment

```shell
Windows: Windows.Desktop v10.0.22622.601
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.19.3531.0
```

**Comment by russellbanks** [comment]
This is because of https://github.com/microsoft/winget-cli/issues/2366.

A temporary solution would be what I did for Android Studio (Stable) - https://github.com/microsoft/winget-pkgs/issues/86933#issuecomment-1296332263

**Comment by denelon** [comment]
How is the version number reported in Windows Apps & Features?

**Comment by russellbanks** [comment]
> How is the version number reported in Windows Apps & Features?

The version number will be reported as `2022.2.1`. However, having it like this would mean winget-pkgs couldn't have all the canary versions, only the latest major ones. This is the case for the other Android Studio channels - see #2366.

**Comment by denelon** [comment]
Does the "displayName" include Canary for the canary versions?

**Comment by russellbanks** [comment]
> Does the "displayName" include Canary for the canary versions?

I don't believe so - there's not really a distinction in terms between the channels in terms of what is written to registry.
